addons/survey/models/survey_user.py

```diff
@@ -461,6 +461,23 @@ def _get_inactive_conditional_questions(self):
                     inactive_questions |= question
         return inactive_questions
 
+    def _get_print_questions(self):
+        """ Get the questions to display :
+              - questions that have been answered
+              - + the ones that should have been answered = active questions
+                In case of session, active questions are based on most voted answers
+        :return: active or answered survey.question browse records
+        """
+        survey = self.survey_id
+        answered_questions = self.user_input_line_ids.mapped('question_id')
```

Comment:
Not sure why you need the "answered questions" as well.
In both use cases (regular survey and most voted answers), once you're removed the questions that are not part of the survey "path", you should get a correct resultset containing all the questions to display.

I understand that the _answer_ of a user can be different from the path the audience has chosen in session mode but it should still not affect the _questions_ resultset once you do "all_questions - inactive_questions".

What am I missing here?

Reply:
nope, you're right!